src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/casenotes/CaseNotesService.kt

```diff
@@ -86,7 +88,14 @@ class CaseNotesService(
     if (caseNoteEvent.wasAmendedInDPS()) {
       runCatching {
         val mapping = mappingApiService.getOrNullByDpsId(dpsCaseNoteId)
-          ?: throw IllegalStateException("Tried to amend an casenote that has never been created")
+          ?: throw IllegalStateException("Tried to amend an casenote that has no mapping")
+        if (mapping.nomisCaseNoteId != caseNoteEvent.additionalInformation.legacyId) {
+          throw IllegalStateException(
+            "Casenote amendment where legacy id does not match mapping: legacy = ${
+              caseNoteEvent.additionalInformation.legacyId
+            }, mapping = ${mapping.nomisCaseNoteId}",
+          )
+        }
```

Comment:
We dont actually need a mapping atm! DPS pass the nomis id through as well (though this may change in the future)